edward/stats/distributions.py

```diff
@@ -9,68 +9,139 @@ class Distribution:
 
         Returns
         -------
         tf.Tensor
-            For univariate distributions, returns a scalar, vector, or
-            matrix corresponding to the size of input. For
-            multivariate distributions, returns a scalar if vector
-            input and vector if matrix input, where each element in
-            the vector evaluates a row in the matrix.
+            If univariate distribution, returns a tensor of same
+            shape as input.
+            If multivariate distribution, returns a tensor of
+            shape[-1] from input: the outer dimension representing the
+            multivariate dimension is collapsed.
+
+        Notes
+        -----
+        x as a 3d or higher tensor is not guaranteed to be supported
+        (for either univariate or multivariate distribution).
         """
         raise NotImplementedError()
 
     def entropy(self):
         """
         Parameters
         ---------
-        params : np.array or tf.Tensor
-            If univariate distribution, can be a scalar or vector.
+        params : float, np.array, or tf.Tensor
+            If univariate distribution, can be a scalar, vector, or
+            matrix.
             If multivariate distribution, can be a vector or matrix.
 
         Returns
         -------
         tf.Tensor
-            For univariate distributions, returns a scalar or vector
-            corresponding to the size of input. For multivariate
-            distributions, returns a scalar if vector input and vector
-            if matrix input, where each element in the vector
-            evaluates a row in the matrix.
+            If univariate distribution, returns a tensor of same
+            shape as input.
+            If multivariate distribution, returns a tensor of
+            shape[-1] from input: the outer dimension representing the
+            multivariate dimension is collapsed.
 
         Notes
         -----
-        SciPy doesn't always enable vector inputs for
-        univariate distributions or matrix inputs for multivariate
-        distributions. This does.
+        The equivalent method in SciPy is not guaranteed to be
+        supported for vector inputs for univariate distributions and
+        matrix inputs for multivariate distributions. This is.
+
+        params as a 3d or higher tensor is not guaranteed to be
+        supported (for either univariate or multivariate
+        distribution).
         """
         raise NotImplementedError()
 
 class Bernoulli:
     def rvs(self, p, size=1):
-        return stats.bernoulli.rvs(p, size=size)
+        """
+        Examples
+        --------
+        >>> x = bernoulli.rvs(p=0.5, size=1)
+        >>> print(x.shape)
+        (1,)
+        >>> x = bernoulli.rvs(p=np.array([0.5]), size=1)
+        >>> print(x.shape)
+        (1, 1)
+        >>> x = bernoulli.rvs(p=np.array([0.5, 0.2]), size=3)
+        >>> print(x.shape)
+        (3, 2)
+        """
```

Comment:
i really like this. i'll endeavor to include this into the `docstrings` down the road